### PR TITLE
Removes the RCD from engi borgs base kit, and replaces it with a module upgrade

### PR DESCRIPTION
**Pull request by 1080pCat**
## What Does This PR Do

More or less the title. removes the RCD from engineering borgs base kit, and adds a new upgrade module that gives it back, with the level&matts requirements as follows: 
materials 6, engineering 5, and power storage 5.
matts: 30000 metal, and 15000 glass. 
Also adds a new var for _adding_ items rather than replacing for borgs.
## Why It's Good For The Game

The rcd is a extremely powerful tool, even more so for cyborgs, since they can deconstruct Rwalls with it, and being able to rapidly space, or un-space a room. on top of all of that, it is a rather lazy, and boring way to fix station damages. while it is fast, and sometimes the only option for somethings, like a SM delam, or something of similar destructive results, ideally RnD would have the required stuff to upgrade a borg to have the rcd. 
## Testing

i ensured the engi borg did not have a rcd on round start, then spawned a upgrade, and put it into a borg testing both non engi, and engi to make sure it wont give janitor a rcd, and making sure that it did actually give the borg the rcd, and also made sure that it was in the mechfab with the correct level requirements, and also deconstructed into one less level for the three types than the requirement. 
## Changelog
:cl:
add: Added a new upgrade module for the engineering borg that gives them a RCD 
tweak: Engineering borgs no longer spawn with the RCD.
tweak: ERT&emaged engi borgs get the RCD upgrade.
/:cl:

## Additional notes
~~While i do see the concern with malf ai, if the ai is playing smartly, and not hacking risky apcs before their borgs have upgrades, i personally think that is more of a player issue rather than a game issue.
And for the borging machine, could fall under the same argument, and i do not entirely see why every new engineering borg would have the need to build a entire department sized room in about a minute, while also being the stunning backbone of the silicon team. 
Although adding a malf power to give borgs rcds may be something that could alleviate some of the concerns some of you may have with this pr.~~

## Edit
I've added the RCD to the emaged module list, mostly to give ert borgs the rcd on spawn, but will also help with malf ai.

### CODE REVIEW

File: code/game/objects/items/robot/robot_upgrades.dm [hunk 1]
```diff
@@ -398,3 +403,12 @@
 	cyborg.floorbuffer = FALSE
 	cyborg = null
 	return ..()
+
+/obj/item/borg/upgrade/RCD
+	name = " R.C.D upgrade"
```
Comment (Contrabang) [inline code/game/objects/items/robot/robot_upgrades.dm:408]:
```suggestion
	name = " R.C.D. upgrade"
```
Minor grammar thing

**Comment by Denthamos** [comment]
This is only for the baseline engie borg right?  Does not effect Syndicate or ERT engineering borgs?

**Comment by GDNgit** [comment]
> This is only for the baseline engie borg right? Does not effect Syndicate or ERT engineering borgs?

it will only affect engineering module borgs, ERT borgs use the same module as engineering borgs but Syndicate saboborgs don't use the same module

**Comment by Christasmurf** [comment]
Can this module be applied to other modules to no effect; or can it not be applied to them at all? 

If it's the first option, you could have ERT borgs spawn with this upgrade by default, so that they get the RCD if choosing engineering borg and otherwise they get nothing. If it's the second option it may be a little harder. Maybe a couple RCD modules in the engineering supply section of CC.

File: code/game/objects/items/robot/robot_upgrades.dm [hunk 2]
```diff
@@ -398,3 +403,12 @@
 	cyborg.floorbuffer = FALSE
 	cyborg = null
 	return ..()
+
+/obj/item/borg/upgrade/RCD
```
Comment (GDNgit) [inline code/game/objects/items/robot/robot_upgrades.dm:407]: could this be lowercase? will look like a define in VS code which is bad

**Comment by 1080pCat** [comment]
> This is only for the baseline engie borg right? Does not effect Syndicate or ERT engineering borgs?

I've added the rcd to the emag_module to account for this oversight. And it should also assist AI while malf. 
I've edited the change log to reflect this addition.

**Comment by procdrone** [comment]
out of curiosity, what happens if you apply both RCD upgrade and an emag (and/or "illegals"), you dont get double RCDs?

**Comment by jughu** [comment]
Just make borg rcd's unable to deconstruct r-walls and add a r-wall upgrade for later borgs are supposed to be better then the crew at the begin

**Comment by 1080pCat** [comment]
> out of curiosity, what happens if you apply both RCD upgrade and an emag (and/or "illegals"), you dont get double RCDs?

Not anymore, thanks for bringing this up

**Comment by Denthamos** [comment]
> > This is only for the baseline engie borg right? Does not effect Syndicate or ERT engineering borgs?
> 
> it will only affect engineering module borgs, ERT borgs use the same module as engineering borgs but Syndicate saboborgs don't use the same module

So ERT engineering borgs would not have an RCD then?  Cannot sign off on that then as ERT engie borg should have an RCD.  ERT engie borg module would need to be separated then if thats the case.  Till then, objection on my part.

**Comment by GDNgit** [comment]
> > > This is only for the baseline engie borg right? Does not effect Syndicate or ERT engineering borgs?
> > 
> > 
> > it will only affect engineering module borgs, ERT borgs use the same module as engineering borgs but Syndicate saboborgs don't use the same module
> 
> So ERT engineering borgs would not have an RCD then? Cannot sign off on that then as ERT engie borg should have an RCD. ERT engie borg module would need to be separated then if thats the case. Till then, objection on my part.

Pr author already changed it so hacked borgs get an rcd (emmaged borgs and ert borgs)

**Comment by Denthamos** [comment]
> > > > This is only for the baseline engie borg right? Does not effect Syndicate or ERT engineering borgs?
> > > 
> > > 
> > > it will only affect engineering module borgs, ERT borgs use the same module as engineering borgs but Syndicate saboborgs don't use the same module
> > 
> > 
> > So ERT engineering borgs would not have an RCD then? Cannot sign off on that then as ERT engie borg should have an RCD. ERT engie borg module would need to be separated then if thats the case. Till then, objection on my part.
> 
> Pr author already changed it so hacked borgs get an rcd (emmaged borgs and ert borgs)

Ah, did not realize the engineering borg counted as 'hacked'.  What a hacky way of making a borg module.  Probably in the future like I mentioned that should be separated to avoid any future nonsense and allow the ERT borg to have more specialized CC stuff.

File: code/game/objects/items/robot/robot_upgrades.dm [hunk 3]
```diff
@@ -398,3 +404,18 @@
 	cyborg.floorbuffer = FALSE
 	cyborg = null
 	return ..()
+
+/obj/item/borg/upgrade/rcd
+	name = " R.C.D. upgrade"
```
Comment (S34NW) [inline code/game/objects/items/robot/robot_upgrades.dm:409]:
```suggestion
	name = "R.C.D. upgrade"
```

**Review by farie82 — APPROVED** [review]

**Pull request merged**